### PR TITLE
Fix link substitution

### PR DESCRIPTION
**Pull request by danieljurek**
Fixes https://github.com/Azure/azure-sdk-for-c/issues/1819 

Current configuration in the default branch does not substitute links properly. 

To verify -- 
1. Visit [example page](https://azuresdkdocs.blob.core.windows.net/$web/c/az_core/1.2.0-dev.2021070801/index.html)
2. ctrl + f "azure core porting guide" 
3. mouse over link 

Actual -- https://github.com/Azure/azure-sdk-for-c/tree/1.2.0-dev.2021070801/sdk/docs/core#porting-the-azure-sdk-to-another-platform

### CODE REVIEW

File: eng/scripts/Language-Settings.ps1 [hunk 1]
```diff
@@ -47,10 +47,9 @@ function Publish-c-GithubIODocs ($DocLocation, $PublicArtifactLocation)
     # Those loops are left over from previous versions of this script which were
     # used to publish multiple docs packages in a single invocation.
     $pkgInfo = Get-Content $DocLocation/package-info.json | ConvertFrom-Json
-    $releaseTag = RetrieveReleaseTag "C" $PublicArtifactLocation
+    $releaseTag = RetrieveReleaseTag -artifactLocation $PublicArtifactLocation
```
Comment (ahsonkhan) [inline eng/scripts/Language-Settings.ps1:50]: I am not very familiar with this script, I am going to assume this is somehow doing the right thing now :)

File: eng/scripts/Language-Settings.ps1 [hunk 2]
```diff
@@ -1,6 +1,6 @@
 $Language = "c"
 $PackageRepository = "C"
-$packagePattern = "*.json"
+$packagePattern = "package-info.json"
```
Comment (ahsonkhan) [inline eng/scripts/Language-Settings.ps1:3]: Can you help me understand, what's this change doing?

**Review by ahsonkhan — APPROVED** [review]

File: eng/scripts/Language-Settings.ps1 (hunk 2 above)
Comment (danieljurek) [reply eng/scripts/Language-Settings.ps1:3]: This fix scopes release info to `package-info.json` files. We ran into an issue in C++ where a new `.json` file was introduced into the place where package release info is loaded. 

While the specific fix for C++ was for vcpkg, this narrowing of the search scope helps prevent potential future issues where a new `.json` file would be introduced. 

C++ fix: https://github.com/Azure/azure-sdk-for-cpp/pull/2330

File: eng/scripts/Language-Settings.ps1 (hunk 1 above)
Comment (danieljurek) [reply eng/scripts/Language-Settings.ps1:50]: This file was missed in a SDK-wide update of release tag retrieval.

File: eng/scripts/Language-Settings.ps1 (hunk 2 above)
Comment (ahsonkhan) [reply eng/scripts/Language-Settings.ps1:3]: Thanks for the pointer, it is a similar change to https://github.com/Azure/azure-sdk-for-cpp/pull/2330

**Pull request merged**